src/lib.rs

```diff
@@ -52,24 +52,20 @@ pub fn frame_size(header: &FrameHeader) -> Option<usize> {
         Bitrate::FreeFormat => None,
 
         Bitrate::Indexed(bitrate) => {
-            let padding_size = match (header.padding, &header.layer) {
-                (true, &Layer::LayerI) => 4_usize,
-                (true, _) => 1_usize,
-                _ => 0_usize,
+            let slot_size = match &header.layer {
+                &Layer::LayerI => 4_usize,
+                _ => 1_usize,
             };
-
-            let samples_per_frame = match header.layer {
-                Layer::LayerI => 384_usize,
-                _ => 1152_usize,
+            let padding = header.padding as usize;
```

Comment:
Maybe also have a `let sampling_rate = header.sampling_rate as usize;` here to make L66 easier to read?